earth2mip/networks/pangu.py

```diff
@@ -253,9 +256,51 @@ def __call__(self, time, x, normalize=False, restart=None):
                 yield time0, x0, restart_data
 
 
+def _download_default_package(
```

Comment:
- Add `PanguPackage`, `DLWPPackage` that does this downloading
- to instantiate, can pass name to `model_registry.ModelRegistry.get_model`.

Reply:
I'll work on this, however, this sacrifices the isolation of model specific code for isolation of IO code. However I agree isolating IO would probably be better for future developments.

Reply:
Okay, so I've moved the NGC ones over into specific Packages but kept Pangu still in its own file...

Looking at this, I really have some reservations about this approach as we are now coupling models with the concept of a package. (I would argue this is worse than having the previous download functions in hindsight now that I see it) So now if I want to add FCN, I now need to update two locations instead of one which is more confusing and messier.

The only info one has to operate with is the name of the model (name of the folder). With the current control  flow, I still think the load function is the best spot for custom download functions like the pangu. I think the real solution is package should now include a `download` function that can be called inside load and pass the specifics onto the file system. Its not exactly `get` because that is for getting a file path inside the package while what we want is a function to get the function itself.

Reply:
I did however remove edits the the base package constructor, even though getting the name of the package is probably a useful property it should be revisited more thoroughly

Reply:
> I really have some reservations about this approach as we are now coupling models with the concept of a package.

I agree, but I suggested this as a temporary work-around that allows adding the feature, without adding new explicit I/O to the load implementations One could envision making a "ZipPackage" or generalizing package to support the http/ngc + .zip case so that we don't need to make a new package class for every model.

The issue we are running into is that no one really built a package of Pangu. The `Package` abstraction exists because internally we had the same model stored on different storage systems, so our load functions could not simply refer to a global address like "https://get.ecmwf.int/repository/test-data/ai-models/pangu-weather/". But when a model is hosted in a globally accessible location like Pangu is, it feels like unneeded work to build a `Package`. One potential solution is that we add support `earth2mip.networks.get_model` to use a loader that does not take a `Package` as a requirement.

However, for now I would be happy to merge this if you create a `PanguPackage` class and refactor as we discussed.

Reply:
> One could envision making a "ZipPackage" or generalizing package to support the http/ngc + .zip case so that we don't need to make a new package class for every model, and I agree this would be better and suggested it in our video chat. but you seemed reticent to take this approach because of time constraints.

This is allowing packages to build themselves which is new. I think we are in an agreement that we should limit model specific stuff to the network file itself. This is very important for making adding models easier and improving developer experience. however this means that the build of the package will still need to get triggered from load. So you've effectively done a full circle, just moving where these urlib/ zip files live but not when they get executed.

Again because you have a single entry point. Alternative is you have a separate entry point for building the package itself but I haven't fully though this through.

Reply:
Pangu package added.